### PR TITLE
ci(pwa): use frozen PNPM lockfile

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Branch?       | main
| Tickets       | n/a
| License       | MIT
| Doc PR        | n/a

Ensure that builds are reproducible.

### CODE REVIEW

File: pwa/Dockerfile [hunk 1]
```diff
@@ -34,7 +34,7 @@ FROM deps as dev
 EXPOSE 3000
 ENV PORT 3000
 
-CMD ["sh", "-c", "pnpm install -r --offline; pnpm dev"]
```
Comment (dunglas) [inline pwa/Dockerfile:37]: In dev, we don't care about reproducibility nor build times.

File: pwa/Dockerfile [hunk 2]
```diff
@@ -25,7 +25,7 @@ COPY --link pnpm-lock.yaml ./
 RUN pnpm fetch
 
 COPY --link . .
-RUN pnpm install -r --offline
+RUN pnpm install --offline --frozen-lockfile
```
Comment (dunglas) [inline pwa/Dockerfile:28]: `-r` is useless here as we don't use workspaces, and is proven to create issues.

**Comment by soyuka** [comment]
Can we use `npm run build` instead of `pnpm build` ? indeed npm also runs post/pre scripts but pnpm doesn't

**Review by soyuka — APPROVED** [review]

**Comment by dunglas** [comment]
@soyuka the goal is to rely on PNPM only and to get rid of NPM at some point.

**Pull request merged**